### PR TITLE
[RLlib] CQL change hparams and data reading strategy

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: avnish <avnish@anyscale.com>

image

tweaking the cql the same way I tweaked pretty much every other offline algorithm so far.
Added the dataset reader for faster speeds. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/12452
merge pending this test passing

**Comment by avnishn** [comment]
buildkite/ray-builders-pr — Build #41560 failed (3 hours, 2 minutes, 32 seconds)
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41560)
buildkite/ray-builders-pr/octopus-brain-tune-tests-and-examples-using-rllib — Failed
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41560#01826ad6-832a-48e7-b9bb-db28942298d8)
buildkite/ray-builders-pr/python-medium-k-z — Failed
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41560#01826ade-1187-4a1b-b33e-ec59f5a3fd36)
buildkite/ray-builders-pr/windows-build-and-test — Failed
[Details](https://buildkite.com/ray-project/ray-builders-pr/builds/41560#01826aa4-d841-42da-9780-543f59412c12)

failing tests unrelated. 
release tests passing:
image

This is ready to merge

File: release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml [hunk 1]
```diff
@@ -25,19 +28,18 @@ cql-halfcheetahbulletenv-v0:
...
         target_network_update_freq: 0
         min_train_timesteps_per_iteration: 1000
         optimization:
             actor_learning_rate: 0.0001
             critic_learning_rate: 0.0003
             entropy_learning_rate: 0.0001
-        num_workers: 0
         num_gpus: 1
         metrics_smoothing_episodes: 5
+        min_time_s_per_iteration: 30
```
Comment (kouroshHakha) [inline release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml:42]: Was this necessary? This is offline RL so more number of iterations should have worked equally here. If this is necessary something does not make sense, if it wasn't necessary we should remove this hparam and instead increase timesteps_total to be consistent with other learning tests.

Comment (avnishn) [reply release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml:42]: wait this parameter controls the logging frequency. I increased it so that we don't run too many unnecessary evaluations

Comment (kouroshHakha) [reply release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml:42]: it's not just that, but also the training time spent per iteration. so you keep running the same training_step() function until that timing requirement is met. In other words you keep taking gradient updates until that time is reached.

Comment (kouroshHakha) [reply release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml:42]: We chatted offline and my concern was more about reproducibility. These nits are not merge blockers. So please merge if it's the only thing holding this back.

**Review by kouroshHakha — APPROVED** [review]

File: release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml (hunk 1 above)
Comment (avnishn) [reply release/rllib_tests/learning_tests/yaml_files/cql/cql-halfcheetahbulletenv-v0.yaml:42]: will open a separate pr that addresses this issue across all release tests and environments
